packages/enzyme-adapter-utils/src/Utils.js

```diff
@@ -260,10 +260,10 @@ export function findElement(el, predicate) {
 }
 
 export function propsWithKeysAndRef(node) {
-  if (node.ref !== null || node.key !== null) {
+  if (node.ref !== null || node.key != null) {
     return {
       ...node.props,
-      key: node.key,
+      key: node.key || null,
```

Comment:
these changes may be fine, or they may need to be moved to the adapter, not sure.